packages/sync/src/modules/class-themes.php

```diff
@@ -795,13 +795,15 @@ private function get_theme_support_info( $theme = null, $send_full_theme_data =
 			$theme = wp_get_theme();
 		}
 
+		if ( $send_full_theme_data ) {
```

Comment:
We need to update line 473 of class-themes.php to pass $send_full_theme_data as true.

Otherwise when switch_theme is triggered only name/version/uri/slug will be synced. When we want to have all the data stay aligned.

Reply:
That is expected. Since the full data is not available yet when theme switch action is fired. Since at that time the theme is not loaded yet. But it should be on the next sync event. Previously we were sending that that but it was wrong.

Reply:
Thanks for that clarification. I'll run some tests on the overall flow and verify the data on the WP.com side this evening.

Reply:
@enejb there is a concern with not passing the full data. If the queue backs up we could sync the the theme options before switch_theme action is processed. This means that the `jetpack_sync_current_theme_support` action will update the `jetpack_global__wp_theme_features` option and leave us without the feature data.

I agree we can't pass the data here as the context is the previous theme. I wonder if we can monitor the send so that if the action includes 'jetpack_sync_current_theme_support` it sends the get_theme_support function data.

Haven't thought through the various ways we can handle this but maybe it should be on the WP.com side where we check the existing value in https://opengrok.a8c.com/source/xref/jetpack/tests/php/sync/server/class.jetpack-sync-server-replicator.php#88 and only call update if the theme name/uri/slug or combination are different.